examples/ping-pong/src/main.rs

```diff
@@ -0,0 +1,118 @@
+    let eth_node = env::var("ETH_NODE").expect("No ETH URL provided.");
+    let provider: Provider<Http> = Provider::<Http>::try_from(eth_node.clone()).unwrap();
+
+    // This can be any string
+    let radio_name: &str = "ping-pong";
+
+    let gossip_agent = GossipAgent::new(
+        private_key,
+        eth_node,
+        radio_name,
+        // Content topics that the Radio subscribes to
+        // If we pass in `None` Graphcast will default to using the ipfs hashes of the subgraphs that the Indexer is allocating to.
+        // But in this case we will override it with something much more simple.
+        Some(vec!["ping-pong-content-topic"]),
+        None,
+        None
+    )
+    .await
+    .unwrap();
+
+    // A one-off setter to load the Gossip Agent into the global static variable
+    _ = GOSSIP_AGENT.set(gossip_agent);
+
+    // A one-off setter to instantiate an empty vec before populating it with incoming messages
+    _ = MESSAGES.set(Arc::new(Mutex::new(vec![])));
+
+    // Helper function to reuse message sending code
+    async fn send_message(content: String, block_number: u64) {
+        match GOSSIP_AGENT
+            .get()
+            .unwrap()
+            .send_message(
+                // The identifier can be any string that suits your Radio logic
+                // If it doesn't matter for your Radio logic (like in this case), you can just use a UUID or a hardcoded string
+                "ping-pong-content-topic".to_string(),
+                block_number,
+                content,
+            )
+            .await
+        {
+            Ok(sent) => println!("Sent message id:: {}", sent),
+            Err(e) => println!("Failed to send message: {}", e),
+        };
+    }
+
+    // The handler specifies what to do with incoming messages.
+    // There cannot be any non-deterministic (this includes async) code inside the handler.
+    // That is why we're saving the message for later processing, where we will check it's content and perform some action based on it.
+    let radio_handler = |msg: Result<GraphcastMessage, anyhow::Error>| match msg {
+        Ok(msg) => {
+            println!("New message received! {:?}\n Saving to message store.", msg);
+            MESSAGES.get().unwrap().lock().unwrap().push(msg);
+        }
+        Err(err) => {
+            println!("{}", err);
+        }
+    };
+
+    GOSSIP_AGENT
+        .get()
+        .unwrap()
+        .register_handler(Arc::new(Mutex::new(radio_handler)));
+
+    loop {
+        let block_number = U64::as_u64(&provider.get_block_number().await.unwrap());
+        println!("🔗 Block number: {}", block_number);
+
+        // If block number is even, send ping message
+        if block_number & 2 == 0 {
+            send_message("Ping".to_string(), block_number).await;
+        };
+
+        // If block number is odd, process received messages
+        if block_number & 2 == 0 {
```

Comment:
probably a typo here - this is conditioning for even numbers, I think you can use `== 1` or `else` with the above `if` statement

Reply:
oh ya 😅 thanks!

Reply:
this should be fixed now